### PR TITLE
Re-enable the retry logic for App Password creation

### PR DESCRIPTION
**Pull request by bmc-msft**
PR #332 inadvertently broke app password creation retry logic.

This re-adds it in a fashion that calling registration.py directly benefits from the retry attempts.

### CODE REVIEW

**Comment by bmc-msft** [comment]
Note, this PR also addresses issues from flake8.  I will look to enabling flake8 in the cicd pipeline in the near future.

File: src/deployment/registration.py [hunk 1]
```diff
@@ -354,7 +376,8 @@ def assign_scaleset_role(onefuzz_instance_name: str, scaleset_name: str) -> None
 
     if not managed_node_role:
         raise Exception(
-            "ManagedNode role not found int the onefuzz application registration. Please redeploy the instance"
+            "ManagedNode role not found int the onefuzz application "
```
Comment (ranweiler) [inline src/deployment/registration.py:379]:
```suggestion
            "ManagedNode role not found in the OneFuzz application "
```

File: src/deployment/data_migration.py [hunk 2]
```diff
@@ -47,7 +47,6 @@ def migrate_notification_keys(table_service: TableService) -> None:
     notifications = table_service.query_entities(
         table_name, select="PartitionKey,RowKey,config"
     )
-    partitionKey = None
```
Comment (ranweiler) [inline src/deployment/data_migration.py:50]: Just unused, I assume?

File: src/deployment/registration.py [hunk 3]
```diff
@@ -211,6 +211,28 @@ def create_application_registration(
 
 
 def add_application_password(app_object_id: UUID) -> Tuple[str, str]:
+    # Work-around the race condition where the app is created but passwords cannot
```
Comment (ranweiler) [inline src/deployment/registration.py:214]: Do we have any generic exponential backoff code we could use in this function?

**Review by ranweiler — APPROVED** [review]

File: src/deployment/data_migration.py (hunk 2 above)
Comment (bmc-msft) [reply src/deployment/data_migration.py:50]: Correct.

File: src/deployment/registration.py (hunk 3 above)
Comment (bmc-msft) [reply src/deployment/registration.py:214]: We don't at this time.  We could use something like that. I'll work on that in the future.

**Pull request merged**